pkg/controller/volume/attachdetach/reconciler/reconciler.go

```diff
@@ -203,9 +208,11 @@ func (rc *reconciler) reconcile() {
 		if rc.actualStateOfWorld.VolumeNodeExists(
 			volumeToAttach.VolumeName, volumeToAttach.NodeName) {
 			// Volume/Node exists, touch it to reset detachRequestedTime
+			// TODO: I don't understand why this is the case.  If we are in the process of detaching something, we "forget" if we desire it to be attached?
 			glog.V(5).Infof("Volume %q/Node %q is attached--touching.", volumeToAttach.VolumeName, volumeToAttach.NodeName)
 			rc.actualStateOfWorld.ResetDetachRequestTime(volumeToAttach.VolumeName, volumeToAttach.NodeName)
 		} else {
+			// TODO-P0: Only do this if the volume is not still attached elsewhere?  Users see the errors and think we are broken, though we aren't.
```

Comment:
this is not an error. Here we check the volume is in desired state, but not in actual state, so reconciler will try to attach volume.